test/unit/modules/helpers/get-n-parents-spec.js

```diff
@@ -28,7 +27,113 @@ describe('RecordingLibrary.helpers.getNParents', () => {
   p2.appendChild(p1);
   p1.appendChild(node);
 
-  test('should return element attributes up 5 parents away if no limit is specified', () => {
+  test(`should return element attributes up 5 parents away if
+    (1) no limit is specified
+    (2) tagName does not include 'button', 'select', 'input', or 'a'
+    (3) role attribute does not include 'button', 'checkbox', 'radio', 'link'`,
+  () => {
+    expect(getNParents(node)).toEqual([
+      {
+        class: 'span-class',
+        id: null,
+        name: null,
+        all_attrs: {
+          class: 'span-class',
+          unique_selector: '.span-class',
+        },
+        node_name: 'SPAN',
+        tag_name: 'SPAN',
+        text: null,
+        title: null,
+        type: null,
+        href: null,
+        nth_parent: 1,
+      },
+      {
+        class: null,
+        id: 'pid',
+        name: null,
+        all_attrs: {
+          id: 'pid',
+          unique_selector: '#pid',
+        },
+        node_name: 'P',
+        tag_name: 'P',
+        text: null,
+        title: null,
+        type: null,
+        href: null,
+        nth_parent: 2,
+      },
+      {
+        class: null,
+        id: null,
+        name: 'divname',
+        all_attrs: {
+          name: 'divname',
+          unique_selector: 'div',
+        },
+        node_name: 'DIV',
+        tag_name: 'DIV',
+        text: null,
+        title: null,
+        type: null,
+        href: null,
+        nth_parent: 3,
+      },
+      {
+        class: 'label-class',
+        id: null,
+        name: null,
+        all_attrs: {
+          class: 'label-class',
+          unique_selector: '.label-class',
+        },
+        node_name: 'LABEL',
+        tag_name: 'LABEL',
+        text: null,
+        title: null,
+        type: null,
+        href: null,
+        nth_parent: 4,
+      },
+      {
+        class: null,
+        id: null,
+        name: null,
+        all_attrs: {
+          title: 'h1title',
+          unique_selector: 'h1',
+        },
+        node_name: 'H1',
+        tag_name: 'H1',
+        text: null,
+        title: 'h1title',
+        type: null,
+        href: null,
+        nth_parent: 5,
+      },
+    ]);
+  });
+
+  test(`should return element attributes up 9 parents away instead of 5 parents away because:
+    (1) no limit is specified and either:
+      (a) tagName includes a 'button', 'select', 'input', 'a' or
+      (b) attribute includes a 'button', 'checkbox', 'radio', 'link'`,
+  () => {
+    const p4WithAnchorTag = document.createElement('a');
+    p4WithAnchorTag.setAttribute('href', '#foo');
+    p4WithAnchorTag.setAttribute('type', 'submit');
+
+    const pSection = document.createElement('section');
+    pSection.setAttribute('class', 'parent-section');
+
+
+    p5.removeChild(p4);
+    p5.appendChild(pSection);
+    pSection.appendChild(p4WithAnchorTag);
```

Comment:
I swapped the label tag here with an anchor tag so we can test the new logic + added in an extra parent so we simulate the full recursive effects when there is a tagName or role attribute that requires us to collect additional parents.

Reply:
if we wanted to, we could add another test that uses a role attribute that triggers the new logic but that seems a bit redundant to me.